### DISCUSSION TITLE
How to make RPG-style text boxes?

### DISCUSSION THREAD

**Discussion by ghost** [dialogic-godot/dialogic#896] [Q&A] [answered]
I'm trying to make a text box like this (from OneShot, play it, very good game!):
![image](https://user-images.githubusercontent.com/25231103/168533149-079fe06a-8732-439d-a505-16e7b85647a0.png)
And this is what I have so far:
![image](https://user-images.githubusercontent.com/25231103/168533307-09eed607-da24-4f4f-9127-7d6abd7c6533.png)

I did this by making a very high offset for the portrait and making portraits appear infront of the textbox.

Is there any better way to do this? (with text wrap)

I did see this PR: https://github.com/coppolaemilio/dialogic/pull/690
But it got closed with only "new features now allow you to do this style of portraits, so I'm closing this pr", not specifying which features.

**Comment by Yotopihi-Dev** [comment]
try changing the Z index of the portrait

**Reply by ghost** [reply]
That is what I did (well not exactly, i set it to show portraits above textboxes), and you can see the result in the 2nd image, as you see, there is no text wrapping.

**Reply by Yotopihi-Dev** [reply]
You should get the latest version from GitHub, it lets you change the margin of a text. Just delete dialogic in the add-ons folder, so it won't delete anything you've added already

**Reply by ghost** [reply]
> You should get the latest version from GitHub, it lets you change the margin of a text. Just delete dialogic in the add-ons folder, so it won't delete anything you've added already

Huh, I didn't know my hacky workaround was an actually acceptable answer?

**Reply by ghost** [reply]
Nevermind, setting the margin does not do anything.

**Reply by Yotopihi-Dev** [reply]
by the way if this doesnt work you can just use a new line so it doesnt overlap

**Reply by Yotopihi-Dev** [reply]
There's a node in dialogic called the "dialognode" You can change how far the label goes

**Reply by Yotopihi-Dev** [reply]
> Nevermind, setting the margin does not do anything.

ah sorry, I thought it'd help

**Reply by ghost** [reply]
It seems like an option to set the text margin (not textbox) exists in the code but is nowhere in the UI...
Nevermind, it exists in the UI as Box padding????? How does that have anything to do with text margin?
Except this has nothing to do with text wrap...

**Reply by zaknafean** [reply]
Are you using the current main branch of Dialogic? A recent patch in our 1.4.2 beta has smoothed over the UI/UX of margins that should help.
![image](https://user-images.githubusercontent.com/7741797/168638890-381cc754-5b0e-48a8-958a-93813624e818.png)
 This along with 'Single Portrait' mode
 
![image](https://user-images.githubusercontent.com/7741797/168639086-0214ef33-d283-49e1-8ea3-2c6480fee3b4.png)

Should get the behavior you're looking for.

**Reply by ghost** [reply]
Thank you, that works! Could you turn your comment into an answer so I can accept it?

**Comment by zaknafean** [comment, marked as answer]
Are you using the current main branch of Dialogic? A recent patch in our 1.4.2 beta has smoothed over the UI/UX of margins that should help.
![image](https://user-images.githubusercontent.com/7741797/168638890-381cc754-5b0e-48a8-958a-93813624e818.png)
 This along with 'Single Portrait' mode
 
![image](https://user-images.githubusercontent.com/7741797/168639086-0214ef33-d283-49e1-8ea3-2c6480fee3b4.png)

Should get the behavior you're looking for.

**Reply by ghost** [reply]
Actually, this still has a problem where if the dialog is hidden, the portrait isn't. You can fix this by making the character leave and join when its hidden. ~~This does break name labels though.~~ Different issue.

**Reply by zaknafean** [reply]
I'm actually confused by this. How are you hiding the dialog but not the portrait? What's the use case you're doing.

**Reply by ghost** [reply]
I'm using the Wait event with Hide DialogBox.

**Reply by zaknafean** [reply]
Oh OK that makes sense.
